### PR TITLE
Fix 3scale smtp flaky test

### PR DESCRIPTION
**Pull request by psturc**
# Description
* scale down "system-app" and "system-sidekiq" in order to load new smtp configuration instead of redeploying them
* use `check3ScaleReplicasAreReady` shared function instead of using the local `checkThreeScaleReplicasAreReady` func that doesn't work in OSD
* [get installtype only once when running functional test suite](https://github.com/integr8ly/integreatly-operator/commit/17187e5e94874c40d7c3092b761c35592ede4202)
* align test user & customer admin user names format with sso-idp script (i.e., test-user0x, customer-admin0x)

# To verify
1. Comment out these lines: https://github.com/integr8ly/integreatly-operator/blob/5ecb3c22a18927b39bd248c07ab6497f31500420/test/functional/integreatly_test.go#L22-L40
2. oc login to OSD cluster with RHOAM installed and SSO IDP configured with the script
```bash
DEDICATED_ADMIN_PASSWORD=Password1 PASSWORD=Password1 ./scripts/setup-sso-idp.sh
```
3. Run the tests
```
INSTALLATION_TYPE=managed-api make test/functional | tee test-results.log
```

### CODE REVIEW

**Comment by KevFan** [comment]
Verified against multi-az release cluster :+1: 
/lgtm
/retest

**Comment by philbrookes** [comment]
/retest
fails due to PROW cluster issues

**Comment by philbrookes** [comment]
@pmccarthy @jjaferson This is being affected by the same fails from the grafana image update PR now.

**Comment by philbrookes** [comment]
I think we should mark E02 and E03 as flaky until the cause is found and addressed.

File: test/common/testing_idp.go [hunk 1]
```diff
@@ -131,7 +131,7 @@ func createTestingIDP(t *testing.T, ctx context.Context, client dynclient.Client
 			return false, fmt.Errorf("failed to create temporary client for idp setup: %w", err)
 		}
 
-		dedicatedAdminUsername := fmt.Sprintf("%s-%d", defaultDedicatedAdminName, defaultNumberOfDedicatedAdmins-1)
+		dedicatedAdminUsername := fmt.Sprintf("%s-%02d", defaultDedicatedAdminName, defaultNumberOfDedicatedAdmins)
```
Comment (KevFan) [inline test/common/testing_idp.go:134]: Extra `-`
```suggestion
		dedicatedAdminUsername := fmt.Sprintf("%s%02d", defaultDedicatedAdminName, defaultNumberOfDedicatedAdmins)
```

**Review by KevFan — COMMENTED** [review]
Looks good :+1: 
Just one small comment

**Comment by KevFan** [comment]
Verified :+1: 
/lgtm

**Comment by philbrookes** [comment]
/retest 
PROW flake

**Pull request merged**